noir-projects/aztec-nr/aztec/src/macros/aztec.nr

```diff
@@ -279,3 +281,26 @@ comptime fn generate_sync_private_state() -> Quoted {
         }
     }
 }
+
+comptime fn generate_process_message() -> Quoted {
+    // We obtain the `utility` function on the next line instead of directly doing
+    // `#[aztec::macros::functions::utility]` in the returned quote because the latter would result in the function
+    // attribute having the full path in the ABI. This is undesirable because we use the information in the ABI only
+    // to determine whether a function is `private`, `public`, or `utility`.
+    let utility = crate::macros::functions::utility;
+
+    quote {
+        #[$utility]
+        unconstrained fn process_message(
+            message: BoundedVec<Field, aztec::messages::encoding::MAX_MESSAGE_LEN>,
+            message_context: aztec::messages::processing::message_context::MessageContext,
```

Comment:
I'm not sure about dealing with plaintext messages. There's an small upside today in that in e.g. the transparent note tests the messages are easy to craft, but that's a red herring since those will soon be replaced by offchain-effects. By having unencrypted messages we show _the app_ some secrets, notably note plaintext, which so far we've never shown: the app typically deals with more processed queries, such as balances, etc. And the app would naturally want to have some encryption happen anyway in whatever transport it uses, so giving it a ciphertext means this is one less thing to worry about for it. @iAmMichaelConnor thoughts?